### DISCUSSION TITLE
Breaking Changes in Minor Versions

### DISCUSSION THREAD

**Discussion by bahrmichael** [sensedeep/dynamodb-onetable#474] [Ideas]
Heya! Just wanted to reach out if there's a strong reason for breaking changes in minor and patch versions (e.g. 2.6.1, 2.6.0, 2.1.1, 2.1.0). Would it be possible to bump the major version when doing breaking changes? That would help a lot for consumers of this package that follow semver.

**Comment by mobsense** [comment]
Yes, thank you for raising this important point. We'll certainly focus on this more.

Can you highlight which changes in particular were breaking for you. That will help our understanding of what kind of changes are slipping through.

**Reply by bahrmichael** [reply]
> Changed the Table.params.partial default to be true

This one from 2.6.0 caused compiler issues that we had to resolve first.

I would suggest just bumping the major version defensively if anything might be a breaking change. Best case consumers see it and think "that's nothing I need to worry about, so let's just increase the version".

**Comment by mobsense** [comment]
Yes, you are right wrt Semver -- https://semver.org/

>Major version X (X.y.z | X > 0) MUST be incremented if any backward incompatible changes are introduced to the public API. It MAY also include minor and patch level changes. Patch and minor versions MUST be reset to 0 when major version is incremented.

We'll do better!

**Reply by bahrmichael** [reply]
Thank you :)
